docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,291 @@
+
+To generate factory data partition using [python script](../../scripts/tools/nrfconnect/nrfconnect_generate_partition.py) use the following command pattern:
+
+```
+$ python nrfconnect_generate_partition.py -i <path_to_JSON_file> -o <path_to_output> --offset <partition_address_in_memory> --size <partition_size> 
+```
+In this command:
+
+- <path_to_JSON_file> is a path to JSON file containing appropriate factory data.
+- <path_to_output> is a path to an output file without any prefix. E.g. By providing here */build/output* as an argument, as a result */build/output.hex* and */build/output.bin* files will be created.
+- <partition_address_in_memory> is an address in the device's persistent storage area where a partition data set should be stored.
+- <partition_size> is a size of partition in the device's persistent storage area. This value will be checked according to JSON data to check if new data fits this size.
+
+To see optional arguments for the script use the following command:
+``` 
+$ python nrfconnect_generate_partition.py -h
+```
+
+**Example of command for nrf52840DK**:
+```
+$ python nrfconnect_generate_partition.py -i build/zephyr/factory_data.json -o build/zephyr/factory_data --offset 0xfb000 --size 0x1000
+```
+
+As a result a *factory_data.hex* and *factory_data.bin* files will be created in */build/zephyr/* directory. The first contains memory offset; therefore, it can be flashed directly to the device using a programmer (E.g. nrfjprog).
+
+
+<hr>
+<a name="Building an example with factory data"></a>
+
+## Building an example with factory data
+
+Generating factory data set can be done manually using instructions described in [Generating factory data](#generating-factory-data) topic. But there is another way to generate factory data. Inside the nrfconnect build system, there is a script that generates factory data automatically using kConfigs and merges it to the firmware .hex file.
+
+To enable generating factory data set automatically build an example with the following option:
+```
+$ west build -b nrf52840dk_nrf52840 -- --DCHIP_FACTORY_DATA_BUILD=y
+```
+or add `CHIP_FACTORY_DATA_BUILD=y` config to a ***prj.conf*** file.
+
+Each factory data parameter has default value and they are described in [Kconfig file](../../config/nrfconnect/chip-module/Kconfig). Setting a new value for factory data parameter can be done in two ways - by providing it as a build argument list or by using interactive Kconfig interfaces.
+
+### Providing factory data parameters as a build argument list
+
+
+This solution of providing factory data can be used with third-party build script because it uses only one command.
+All parameters can be edited manually by providing them as an additional option for the west command e.g.:
+
+```
+$ west build -b nrf52840dk_nrf52840 -- --DCHIP_FACTORY_DATA_BUILD=y --DCHIP_DEVICE_DISCRIMINATOR=0xF11
```

Comment:
Please verify in all commands in this file that:
1. You pass configs with single - and not -- in the command
2. Every one of them has CONFIG_ prefix